Include/cpython/compile.h

```diff
@@ -69,14 +69,14 @@ typedef struct {
 PyAPI_FUNC(int) PyCompile_OpcodeStackEffect(int opcode, int oparg);
 PyAPI_FUNC(int) PyCompile_OpcodeStackEffectWithJump(int opcode, int oparg, int jump);
 
-PyAPI_FUNC(int) PyUnstable_OpcodeIsValid(int opcode);
-PyAPI_FUNC(int) PyUnstable_OpcodeHasArg(int opcode);
-PyAPI_FUNC(int) PyUnstable_OpcodeHasConst(int opcode);
-PyAPI_FUNC(int) PyUnstable_OpcodeHasName(int opcode);
-PyAPI_FUNC(int) PyUnstable_OpcodeHasJump(int opcode);
-PyAPI_FUNC(int) PyUnstable_OpcodeHasFree(int opcode);
-PyAPI_FUNC(int) PyUnstable_OpcodeHasLocal(int opcode);
-PyAPI_FUNC(int) PyUnstable_OpcodeHasExc(int opcode);
+PyAPI_FUNC(int) _PyCompile_OpcodeIsValid(int opcode);
+PyAPI_FUNC(int) _PyCompile_OpcodeHasArg(int opcode);
+PyAPI_FUNC(int) _PyCompile_OpcodeHasConst(int opcode);
+PyAPI_FUNC(int) _PyCompile_OpcodeHasName(int opcode);
+PyAPI_FUNC(int) _PyCompile_OpcodeHasJump(int opcode);
+PyAPI_FUNC(int) _PyCompile_OpcodeHasFree(int opcode);
+PyAPI_FUNC(int) _PyCompile_OpcodeHasLocal(int opcode);
+PyAPI_FUNC(int) _PyCompile_OpcodeHasExc(int opcode);
```

Comment:
Would it be possible to move them to the internal C API if they should not be used outside Python? Also, would it possible to replace PyAPI_FUNC() with extern, to not export them? You can keep PyAPI_FUNC() if they are needed by shared C extensions.

Reply:
They are used in Module/_opcode.c so I'm not sure.

Reply:
Ah maybe it's possible, let me try.

Reply:
I'm getting problems like this when I try to make them extern:

```
[ERROR] _asyncio failed to import: dlopen(/Users/iritkatriel/src/cpython-654/build/lib.macosx-14.1-x86_64-3.13-pydebug/_opcode.cpython-313d-darwin.so, 0x0002): symbol not found in flat namespace '__PyOpcode_HasArg'
[ERROR] _opcode failed to import: dlopen(/Users/iritkatriel/src/cpython-654/build/lib.macosx-14.1-x86_64-3.13-pydebug/_opcode.cpython-313d-darwin.so, 0x0002): symbol not found in flat namespace '__PyOpcode_HasArg'
Following modules built successfully but were removed because they could not be imported:
_asyncio              _opcode                                  

```

Reply:
Ah right, _PyOpcode_HasArg() should be exported for _opcode.

Reply:
I checked: all of these _PyCompile_OpcodeXXX() functions are used by _opcode. Just please just add a comment to say that they are exported for _opcode which is built as a shared extension.

Reply:
All but the last two I can move, but for the last two I get:

```
[ERROR] _asyncio failed to import: dlopen(/Users/iritkatriel/src/cpython-654/build/lib.macosx-14.1-x86_64-3.13-pydebug/_opcode.cpython-313d-darwin.so, 0x0002): symbol not found in flat namespace '__PyCompile_GetBinaryIntrinsicName'
[ERROR] _opcode failed to import: dlopen(/Users/iritkatriel/src/cpython-654/build/lib.macosx-14.1-x86_64-3.13-pydebug/_opcode.cpython-313d-darwin.so, 0x0002): symbol not found in flat namespace '__PyCompile_GetBinaryIntrinsicName'
```

Reply:
Ah right, you should include the internal header file in `Python/intrinsics.c` to export the function:

```diff
diff --git a/Python/intrinsics.c b/Python/intrinsics.c
index a9915ffb69..cc332ea5dc 100644
--- a/Python/intrinsics.c
+++ b/Python/intrinsics.c
@@ -2,6 +2,7 @@
 #define _PY_INTERPRETER
 
 #include "Python.h"
+#include "pycore_compile.h"       // export _PyCompile_GetUnaryIntrinsicName()
 #include "pycore_frame.h"
 #include "pycore_function.h"
 #include "pycore_global_objects.h"
```

Reply:
got it!